src/modules/tokens.ts

```diff
@@ -20,7 +20,7 @@ export namespace tokens {
       // creates a new token and fill properites
       token = new Token(tokenId)
       token.address = tokenAddress
-      token.firstTradeTimestamp = timestamp
+      token.firstTradeTimestamp = timestamp ? timestamp : 0;
```

Comment:
why this? 

timestamp is a i32 already right?  why would u need to do this check? 
Also, you create a token if it there was a trade, so u should have a date for it.